dali/operators/reader/loader/video/frames_decoder_gpu.cc

```diff
@@ -40,28 +40,63 @@ int process_picture_decode(void *user_data, CUVIDPICPARAMS *picture_params) {
 }
 }  // namespace detail
 
+
+void FramesDecoderGpu::InitBitStreamFilter() {
+  const AVBitStreamFilter *bsf = nullptr;
+
+  switch (av_state_->codec_->id) {
+  case AVCodecID::AV_CODEC_ID_H264:
+    bsf = av_bsf_get_by_name("h264_mp4toannexb");
+    break;
+  case AVCodecID::AV_CODEC_ID_HEVC:
+    bsf = av_bsf_get_by_name("hevc_mp4toannexb");
+    break;
+  default:
+    DALI_FAIL(make_string(
+      "Could not find suitable bit stream filter for codec: ",
+      av_state_->codec_->name));
+  }
+
+  DALI_ENFORCE(av_bsf_alloc(bsf, &bsfc_) >= 0);
+  DALI_ENFORCE(avcodec_parameters_copy(
+    bsfc_->par_in, av_state_->ctx_->streams[0]->codecpar) >= 0);
+  DALI_ENFORCE(av_bsf_init(bsfc_) >= 0);
+}
+
+cudaVideoCodec FramesDecoderGpu::FindCodecType() {
+  switch (av_state_->codec_->id) {
+  case AV_CODEC_ID_H264:
+    return cudaVideoCodec_H264;
+  case AV_CODEC_ID_HEVC:
+    return cudaVideoCodec_HEVC;
+  default:
+    DALI_FAIL(make_string(
+      "Unsupported codec: ",
+      av_state_->codec_->name));
+  }
+  return cudaVideoCodec_HEVC;
```

Comment:
This code is unreachable, right?

Reply:
Yes. I refactored the whole function, since the `default` was also dead code. This is checked in the base class constructor. I simplified this function, since we know that everything was checked before. I did it in the simplest way possible. When I add more codecs I will figure out some more elegant way of mapping `AVCoodecID` to `cudaVideoCodec`.